### PR TITLE
Feature/core batch search endpoint

### PR DESCRIPTION
**Pull request by aristotelisxs**
Extending the core_search function's capabilities to include multiple search queries in case results need to be retrieved from a wide range of requests in a more efficient manner.

## Description

Adding in a POST request to the core_search function and a convenience function to the zzz.R file that will convert a simple character vector, e.g. c("data mining", "machine learning", "semantic web"), to a suitable JSON object required from the endpoint to perform a batch search request. This includes updating of the documentation with the added parameter to the function (called "method" for consistency with the similar core_articles function's parameter) and checks to disallow multiple queries in GET requests for the core_search function.

Moreover, single valued parameters in the JSON objects sent to the endpoints will be unboxed as this is a requirement for all of CORE's endpoints.

## Example

As included in the documentation:

query <- c("data mining", "machine learning", "semantic web")
res <- core_search(query, method = "POST")
head(res$data)
res$data$id

The "page" and "pageSize" variables will be constant in all individual query requests (may not be the desirable functionality for some but done so for the sake of simplicity).

### CODE REVIEW

**Comment by sckott** [comment]
thanks @aresxs91 

First, a question. Would it be easier to instead of having a new parameter `method` to use `POST` when `length(query) > 1` and `GET` otherwise ?

**Comment by aristotelisxs** [comment]
Hello Scott,

Thanks for your comment. I see, yes that would indeed make it more concise.
I copied your style for the core_articles function there to be honest. I'll
revise it and let you know :)

Best,
Aris

**Comment by sckott** [comment]
@aresxs91 ping me when you're done

**Comment by aristotelisxs** [comment]
It seems alright from my end, I haven't made extensive tests for the function but it looks ready.

File: R/core_search.R [hunk 1]
```diff
@@ -28,6 +28,17 @@ core_search <- function(query, page = 1, limit = 10, key = NULL,
 #' @rdname core_search
 core_search_ <- function(query, page = 1, limit = 10, key = NULL, ...) {
   must_be(limit)
-  core_GET(path = file.path("search", query), key,
-           list(page = page, pageSize = limit), ...)
+  switch(
```
Comment (sckott) [inline R/core_search.R]: since there's only two options, an if/else makes more sense here i think

File: R/zzz.R [hunk 2]
```diff
@@ -25,7 +25,7 @@ core_POST <- function(path, key, args, body, ...){
   temp <- cli$post(
     path = file.path("api-v2", path),
     query = cp(args),
-    body = jsonlite::toJSON(body), encode = "json", ...
+    body = jsonlite::toJSON(body, auto_unbox=T), encode = "json", ...
```
Comment (sckott) [inline R/zzz.R]: please use `TRUE` and `FALSE` instead of `T` and `F`

File: R/zzz.R [hunk 3]
```diff
@@ -74,3 +74,14 @@ clog <- function(x){
 pdf_parse <- function(x, parse) {
   if (parse) pdftools::pdf_text(x) else x
 }
+
+create_batch_query_list <- function(queries, page, pageSize) {
+  queryList <- lapply(queries, function(x){
+    as.list(setNames(x, rep("query", length(x))))
```
Comment (sckott) [inline R/zzz.R]: namespace `setNames` here like `stats::setNames` since we don't import the stats pkg

**Review by sckott — CHANGES_REQUESTED** [review]
just a few changes

File: R/core_search.R (hunk 1 above)
Comment (aristotelisxs) [reply R/core_search.R]: Agreed.

**Comment by sckott** [comment]
thanks very much!

**Review by sckott — APPROVED** [review]

**Pull request merged**